db/migrate/20150923125825_add_job_invocation_task_group.rb

```diff
@@ -0,0 +1,9 @@
+class AddJobInvocationTaskGroup < ActiveRecord::Migration
+  def up
+    add_column :job_invocations, :task_group_id, :integer, :index => true
```

Comment:
AFAIK you can have foreign key on nullable fields -  it should check the contraint just on the case, when the field is entered.